src/master/validation.cpp

```diff
@@ -298,7 +299,9 @@ Option<Error> reregisterSlave(
     const vector<FrameworkInfo>& frameworkInfos)
 {
   hashset<FrameworkID> frameworkIDs;
-  hashset<ExecutorID> executorIDs;
+
+  typedef pair<FrameworkID, ExecutorID> IDPair;
+  hashset<IDPair> executorIDs;
```

Comment:
Mesos doesn't use convenience typedefs.
```C
hashset<pair<FrameworkID, ExecutorID>> executorIDs
```

In subsequent lines, I believe that this is ok:
```
auto id = std::make_pair(executor.framework_id(), executor.executor_id());
```